libguard/guard_interface.cpp

```diff
@@ -260,14 +260,19 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType,
     return getHostEndiannessRecord(guard);
 }
 
-GuardRecords getAll()
+GuardRecords getAll(bool persistentTypeOnly)
 {
     GuardRecords guardRecords;
     GuardRecord curRecord;
     int pos = 0;
     GuardFile file(guardFilePath);
     for_each_guard(file, pos, curRecord)
     {
+        if (persistentTypeOnly && (curRecord.errType == GARD_Reconfig ||
```

Comment:
you want to fetch if persistentTypeOnly but the condition above looks like if flag set ignore

Reply:
you did not capture in comments about checking for reconfig and sticky deconfig? what is that for

Reply:
How do we determine "the ephemeral type records are  used for internal purpose"

Reply:
is this condition saying is persistent and ephemeral ? with those two checks?